stream-chat-android-ui-components/src/main/res/layout/stream_message_input.xml

```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:background="@color/stream_white">
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:gravity="center_vertical"
+        android:orientation="horizontal"
+        android:paddingTop="6dp"
+        android:paddingBottom="6dp">
+
+        <androidx.appcompat.widget.AppCompatImageView
+            android:id="@+id/iv_open_attachment"
+            android:layout_width="@dimen/stream_attachment_button_width"
+            android:layout_height="@dimen/stream_attachment_button_height"
+            android:layout_marginStart="8dp"
+            android:layout_marginEnd="8dp"
+            android:src="@drawable/stream_ic_attach"
+            tools:src="@drawable/stream_ic_attach" />
+
+        <androidx.appcompat.widget.AppCompatImageView
+            android:id="@+id/iv_open_emojis"
+            android:layout_width="@dimen/stream_attachment_button_width"
+            android:layout_height="@dimen/stream_attachment_button_height"
+            android:layout_marginStart="8dp"
+            android:layout_marginEnd="8dp"
+            android:src="@drawable/stream_ic_lightning"
+            tools:src="@drawable/stream_ic_lightning" />
+
+        <androidx.appcompat.widget.AppCompatEditText
+            android:id="@+id/et_message_text_input"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center"
+            android:layout_weight="1"
```

Comment:
I think LinearLayout can be removed and weights can be implemented within `ConstraintLayout`

Reply:
ConstraintLayout renders slower than LinearLayout, no? If we stack many of then, the view may time time to render itself.

Reply:
Oh, you mean use the root ContraintLayout, right?

Reply:
You're right, makes more sense to use a group of views inside the ConstraintLayout than to use a LinearLayout.